src/task_queue/database_tasks/contents_spacing_correction.py

```diff
@@ -0,0 +1,133 @@
+PATTERN = "|".join(re.escape(p) for p in PATTERNS)
+
+"""
+This File is used to fix spacing for both universal contents(contents not tied to a community) and content tied to a community.
+There are two parts to this:
+1. Generate a report of the contents that need to be fixed with the following information
+    a. Community: the community the content belongs to
+    b. Content type: whether content is an Action/Event/Testimonial etc.
+    c. Item name: the name or title of the content
+    d. Field name: the field name of the content to be corrected
+    e. Count: the number of occurrences of spacing in the content.
+2. Fix the spacing in the database.
+"""
+
+def write_to_csv(data):
+    response = HttpResponse(content_type="text/csv")
+    writer = csv.DictWriter(response, fieldnames=["Community", "Content Type", "Item Name", "Field Name", "Count"])
+    writer.writeheader()
+    for row in data:
+        writer.writerow(row)
+    return response.content
+def get_community(instance):
+    if instance and hasattr(instance, "community"):
+        return instance.community.name if instance.community else "N/A"
+    
+
+def get_model_instances(model_name, app_label):
+    model = apps.get_model(app_label=app_label, model_name=model_name)
+    filter_args = {} if model_name == "PastEvent" else {"is_deleted": False}
+    model_instances = model.objects.filter(**filter_args)
+    return model,model_instances
+
+def get_table_report(field_name, model_name, app_label):
+    data = []
+    try:
+        model, model_instances = get_model_instances(model_name, app_label)
+        for instance in model_instances:
+            field_value = getattr(instance, field_name)
+            if field_value:
+                count = len(re.findall(PATTERN, field_value))
+                if count > 0:
+                    data.append({
+                        "Community": get_community(instance),
+                        "Content Type": model_name,
+                        "Item Name": instance.name if hasattr(model, "name") else instance.title,
+                        "Field Name": field_name,
+                        "Count": count,
+                    })
+    except LookupError as e:
+        print("==ERROR==", e)
+    return data
+
+
+
+def auto_correct_spacing(field_name, model_name, app_label, enabled_communities):
+    try:
+        model, model_instances = get_model_instances(model_name, app_label)
+        for instance in model_instances:
+            if hasattr(instance, "community"):
+                if instance.community in enabled_communities:
+                    field_value = getattr(instance, field_name)
+                    if field_value:
+                        count = len(re.findall(PATTERN, field_value))
+                        if count > 0:
+                            setattr(instance, field_name, field_value.replace("<p><br></p>", "").replace("<p>.</p>", "").replace("<p>&nbsp;</p>", ""))
```

Comment:
You have these patterns defined in PATTERNS, but then they are hardcoded here.  Could you loop over the patterns and do the replace that way?